us-geological-survey.csl

```diff
@@ -195,6 +230,84 @@
           <text variable="title" prefix=" " suffix=":"/>
           <text variable="event" prefix=" "/>
           <text variable="event-place" prefix=", "/>
+          <choose>
+            <if variable="URL">
+              <group prefix=", ">
+                <text term="accessed"/>
+                <group prefix=" " suffix=",">
+                  <date variable="accessed">
+                    <date-part name="month" form="long" suffix=" "/>
+                    <date-part name="day" form="numeric" suffix=", "/>
+                    <date-part name="year" form="long"/>
+                  </date>
+                </group>
+                <group prefix=" ">
+                  <text variable="URL" prefix="at "/>
+                </group>
+              </group>
+            </if>
+          </choose>
+        </else-if>
+        <else-if type="dataset">
+          <text variable="title" prefix=" " suffix=":"/>
+          <text variable="publisher" prefix=" " suffix=", "/>
+          <group prefix=" ">
+            <text term="accessed"/>
+            <group prefix=" " suffix=",">
+              <date variable="accessed">
+                <date-part name="month" form="long" suffix=" "/>
+                <date-part name="day" form="numeric" suffix=", "/>
+                <date-part name="year" form="long"/>
+              </date>
+            </group>
+            <group prefix=" ">
+              <text term="at"/>
+              <text variable="container-title" prefix=" "/>
+            </group>
+            <group prefix=" ">
+              <choose>
+                <if variable="DOI">
+                  <text variable="DOI" prefix="at https://doi.org/"/>
+                </if>
+                <else>
+                  <text term="at"/>
+                  <text variable="URL" prefix=" "/>
+                </else>
+              </choose>
+            </group>
+          </group>
+        </else-if>
+        <else-if type="software">
+          <group suffix=":">
```

Comment:
You should almost never include punctuation like this as a prefix or suffix, as it will show even if there is no text following. Use "delimiter" instead.

Reply:
The format is `[title] [version]:`. USGS style is to either include the version in the title or include it in the version field. I want the colon after version if both are given and after the title if just the title is given. Isn't it safe to require a title?

Reply:
You should never assume that any piece of data is present and always write the style to be robust to missing data. An untitled piece of software might have just a medium/genre and no title for example.